include/xten/Dialect/XTenNN/IR/XTenNNOps.td

```diff
@@ -251,6 +251,42 @@ def XtenNN_MishOp: XTenNN_Op<"mish", [Pure, TosaExtension, ElementwiseUnary, Sam
   let assemblyFormat = [{ operands attr-dict `:` functional-type(operands, results) }];
 }
 
+def XtenNN_ResizeOp: XTenNN_Op<"resize", [Pure, TosaExtension]> {
+  let summary = "Performs an upscale of 2 the the image.";
+  let description = [{
+    Performs an upscale of a factor of 2 to the input image. Some constrains and information:
+      - It does not perform 'antialias' for linear mode (ONNX's antialias==0)
+      - It performs the resize in all dimensions (ONNX's axes==None)
+      - No sampling is perfomed outside the tensor (ONNX's exclude_outside==0)
+      - The input is always stretched to meet the upscale of factor 2 (ONNX's keep_aspect_ratio_policy==stretch)
+      - Cordinate Trasnformation Mode:
+        - 0: half_pixel
+        - 1: asymmetric
+        - 2: align_corners
+      - Mode:
+        - 0: Nearest
+        - 1: Linear
+      - Nearest mode:
+        - 0: floor
+        - 1: round_prefer_ceil
+        - 2: round_prefer_floor
+  }];
+  let arguments = (ins
+    4DTensorOf<[BF16, F32]>:$X,
+    I64Attr:$coordinate_transformation_mode,
+    I64Attr:$mode,
+    I64Attr:$nearest_mode
```

Comment:
Could there be a way for us to compute and annotate the "scales" attributes on the op? Since they are needed in pattern matching, that would avoid us needing a callback do a computation for it

Reply:
Done